janusgraph-server/src/main/java/org/janusgraph/graphdb/grpc/JanusGraphContextHandler.java

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.graphdb.grpc;
+
+import org.apache.tinkerpop.gremlin.server.GraphManager;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.janusgraph.core.JanusGraph;
+import org.janusgraph.graphdb.grpc.schema.SchemaManagerProvider;
+import org.janusgraph.graphdb.grpc.types.JanusGraphContext;
+
+import java.util.ArrayList;
+
+public class JanusGraphContextHandler {
+    private final GraphManager graphManager;
+
+    public JanusGraphContextHandler(GraphManager graphManager) {
+        this.graphManager = graphManager;
+    }
+
+    public Iterable<JanusGraphContext> getContexts() {
+        ArrayList<JanusGraphContext> contexts = new ArrayList<>();
+        for (String graphName : graphManager.getGraphNames()) {
+            Graph graph = graphManager.getGraph(graphName);
+            if (!(graph instanceof JanusGraph)) {
+                continue;
+            }
+            contexts.add(JanusGraphContext.newBuilder().setGraphName(graphName).build());
+        }
+        return contexts;
+    }
+
+    public JanusGraphContext getContextByGraphName(String graphName) {
+        Graph graph = graphManager.getGraph(graphName);
+        if (!(graph instanceof JanusGraph)) {
+            return null;
+        }
+        return JanusGraphContext.newBuilder().setGraphName(graphName).build();
+    }
+
+    public SchemaManagerProvider getSchemaManagerProviderByContext(JanusGraphContext context) {
+        Graph graph = graphManager.getGraph(context.getGraphName());
+        if (!(graph instanceof JanusGraph)) {
+            return null;
+        }
+        return new SchemaManagerProvider((JanusGraph) graph);
```

Comment:
Do we need to create a new SchemaManagerProvider each time the RPC server receives a request?

Reply:
I don't think we need to create SchemaManagerProvider on each call. 
Keep in mind if we are doing it differently, we have to handling the GraphManager correctly. They can be dynamic register new graphs and remove them.

My idea was to keep the logic as simple as possible.

Reply:
You think or you don't think? If I am understanding it correctly, you are creating a new SchemaManagerProvider on each call.

Reply:
At moment I would keep as simple as possible in the future would like to cache the provider some how.